src/ui_interface.h

```diff
@@ -95,6 +97,9 @@ class CClientUIInterface
 
     /** New block has been accepted */
     boost::signals2::signal<void (const uint256& hash)> NotifyBlockTip;
+
+    /** New transaction to relay */
+    boost::signals2::signal<void (const CTransaction& tx)> NotifyRelayTx;
```

Comment:
I don't think NotifyRelayTx should exist at all.
As suggested by @jonasschnelli, instead of using CClientUIInterface signals we could use CValidationInterface SetBestChain and SyncTransaction